### PR TITLE
docs: improve troubleshooting and maintenance of a cluster

### LINKED ISSUE #6198: docs: how to shutdown and bring up two nodes in a cluster of three nodes
**Is your feature request related to a problem? Please describe.**
The process to shutdown and bring up **two** nodes in a cluster of three nodes is not documented. 

Currently, we only described how to shutdown and bring back **all** cluster nodes in Maintenance and Operations section.

**Describe the solution you'd like**
It would be nice to have a dedicated section in Cluster guide under Maintenance and Operations which explain how to shutdown and bring up **two** nodes in a cluster of three nodes.

Of course, we need to document how to move service to remaining node.

**Additional context**
Upgrade process can be used as a base because it's a similar process where you put A and B node in maintenance and move service to node C.

### PR DESCRIPTION
**Pull request by nqb**
# Description
Improve troubleshooting and maintenance of a cluster.

# Impacts
Cluster documentation.

# Issue
fixes #6198 

# Delete branch after merge
YES

### CODE REVIEW

**Comment by nqb** [comment]
Once this one is merged, @cgx will update website documentation with this change.

**Comment by nqb** [comment]
@julsemaan,

On a 10.3 cluster setup, all my nodes have a `/var/lib/mysql/grastate.dat` file which contains: `seqno:   -1`.
However, I see a `/var/lib/mysql/galera-autofix-seqno.dat` file with value identical to `wsrep_last_commited`.

So my question is, when you are in the scenario where there is no database service, which file do you need to check on each node to identify the most recent node ?

Thanks.

**Comment by julsemaan** [comment]
You can run MariaDB with --wsrep-recover to get grastate.dat updated or rely on galera-autofix-seqno.dat

**Comment by nqb** [comment]
@julsemaan, I push some changes, let me know if it's ok.

File: docs/cluster/maintenance_and_operations.asciidoc [hunk 1]
```diff
@@ -18,13 +18,21 @@ endif::[]
...
 //== Maintenance and Operations
 
-=== Putting a node in maintenance
+=== Putting nodes in maintenance
 
-When doing maintenance on a node (especially the management node), it is
-always better to put it in maintenance mode so it doesn't try to join an
-existing cluster.
+When doing maintenance on a cluster, it is always better to put nodes
+concerned in a maintenance mode so they doesn't try to join an existing
```
Comment (julsemaan) [inline docs/cluster/maintenance_and_operations.asciidoc]: to put the targeted nodes in a maintenance mode

File: docs/cluster/maintenance_and_operations.asciidoc [hunk 2]
```diff
@@ -18,13 +18,21 @@ endif::[]
...
+cluster. You can achieve this goal using two methods.
```
Comment (julsemaan) [inline docs/cluster/maintenance_and_operations.asciidoc]:
```suggestion
cluster. You can achieve this using two methods.
```

File: docs/cluster/maintenance_and_operations.asciidoc [hunk 3]
```diff
@@ -18,13 +18,21 @@ endif::[]
...
 
-Once node is in maintenance, you can safely reboot it.
+IMPORTANT: In a three nodes cluster, you can offer service with at least one node.
+
+==== Using a clean shutdown
+
+If you stop `packetfence-mariadb` service properly on a node or if you
```
Comment (julsemaan) [inline docs/cluster/maintenance_and_operations.asciidoc]:
```suggestion
If you stop the `packetfence-mariadb` service properly on a node or if you
```

File: docs/cluster/maintenance_and_operations.asciidoc [hunk 4]
```diff
@@ -18,13 +18,21 @@ endif::[]
...
+shutdown your node properly, cluster will detect this shutdown and continue to
```
Comment (julsemaan) [inline docs/cluster/maintenance_and_operations.asciidoc]:
```suggestion
shutdown your node properly, the cluster will detect this shutdown and continue to
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 5]
```diff
@@ -35,9 +35,22 @@ Important variables:
 
 In order for the cluster to be considered healthy, all nodes must be listed under `wsrep_incoming_addresses` and `wsrep_local_state_comment` must be `Synced`. Otherwise look in the MariaDB log ([filename]`/usr/local/pf/logs/mariadb_error.log`)
 
-=== Rejoining a node to an existing cluster
+=== Automatic clustering resolution issue: galera-autofix service
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
=== Automatic clustering resolution service: galera-autofix
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 6]
```diff
@@ -35,9 +35,22 @@ Important variables:
...
 
-When only one node has left cluster, you can apply following commands on that node to rejoin:
+Since v10, the <<_the_galera_autofix_service,`galera-autofix`>> service will try to resolve automatically issues on Galera clusters.
+
+On a cluster which has an issue, once all nodes are back online, you should wait:
+
+* around 10 minutes when at least one of the nodes of the cluster is able to offer database service
+* around 20 minutes when there is no database service available
+
+If your cluster still have issue after that time, you can try to solve it manually by looking at sections below.
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
If your cluster still has issues after that time, you can try to resolve the issue by looking at sections below.
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 7]
```diff
@@ -35,9 +35,22 @@ Important variables:
...
+
+=== Cluster offer database service without all nodes
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
=== Cluster offers database service without all nodes
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 8]
```diff
@@ -50,42 +63,89 @@ systemctl start packetfence-galera-autofix
 
 This action will not cause service disruption on current cluster.
 
-=== Recovering when a node is missing
-
-If one of the nodes cannot recover, you can manually reset the cluster state. Note that this procedure is only valid if all the nodes were hard-shutdown at the same time.
+After all nodes have joined back cluster, you should verify <<_checking_the_mariadb_sync,MariaDB sync>>.
 
-First, stop `packetfence-mariadb` on all servers:
+WARNING: After stopping `packetfence-mariadb` service, be sure there is no more `mysql` process running.
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
WARNING: After stopping the `packetfence-mariadb` service, be sure there is no more `mysql` process running.
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 9]
```diff
@@ -50,42 +63,89 @@ systemctl start packetfence-galera-autofix
...
 
-  # systemctl stop packetfence-mariadb
+=== No more database service
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
=== None of the nodes is offering database service
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 10]
```diff
@@ -50,42 +63,89 @@ systemctl start packetfence-galera-autofix
...
 
-Then remove `/var/lib/mysql/gvwstate.dat` on all the servers that are still alive.
+When there is no more database service in your cluster, you need to do a full recovery.
 
-Next, see the section 'Recovering from a split brain' to start up a new cluster.
+You must identify the node you wish to keep the data from and start it with the
+`--force-new-cluster` option.
 
-=== Recovering from a split brain
+Node which has the most recent data will have a highest `seqno` value in
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
Find the node which has the highest `seqno` value in
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 11]
```diff
@@ -50,42 +63,89 @@ systemctl start packetfence-galera-autofix
...
+[filename]`/var/lib/mysql/grastate.dat`.
 
-If you leave the cluster in auto-recovery mode without electing a master manually, the structure of the cluster is made so that a split-brain can never occur since a server will fallback in read-only if it can't join a primary view (quorum).
+If the `seqno` value is equals to `-1`, you need to start MariaDB manually with
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
If the `seqno` value is `-1`, you need to start MariaDB manually with
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 12]
```diff
@@ -50,42 +63,89 @@ systemctl start packetfence-galera-autofix
...
+`--wsrep-recover` to update `seqno` value:
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
`--wsrep-recover` to update the `seqno` value using the commands below:
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc [hunk 13]
```diff
@@ -50,42 +63,89 @@ systemctl start packetfence-galera-autofix
...
+----
+
+You should then see `/var/lib/mysql` be populated again with the data and once
+MariaDB becomes available again on the server, it means the sync has
+completed. In case of issues, look in the MariaDB log file
+(`/usr/local/pf/logs/mariadb_error.log`)
+
+WARNING: After stopping `packetfence-mariadb` service, be sure there is no more `mysql` process running.
+
+==== On node started with --force-new-cluster
```
Comment (julsemaan) [inline docs/cluster/troubleshooting_a_cluster.asciidoc]:
```suggestion
==== On the node started with --force-new-cluster
```

File: docs/cluster/troubleshooting_a_cluster.asciidoc (hunk 10 above)
Comment (nqb) [reply docs/cluster/troubleshooting_a_cluster.asciidoc]: I could be wrong but your suggestion looks weird to me.

Comment (julsemaan) [reply docs/cluster/troubleshooting_a_cluster.asciidoc]: > I could be wrong but your suggestion looks weird to me.

Yep, just adjusted that

**Comment by nqb** [comment]
All typos fixaid.

**Comment by julsemaan** [comment]
@extrafu a.k.a. the doc king pin

Do you want to take a look at it before I press merge ?

**Comment by extrafu** [comment]
I sure will!

File: docs/cluster/appendix.asciidoc [hunk 14]
```diff
@@ -414,7 +414,13 @@ systemctl restart packetfence-galera-autofix
 
 ==== Maximum connections reached
 
-In the event that one of the 3 servers reaches the maximum amount of connections (defaults to 1000), this will dead-lock the Galera cluster synchronization. In order to resolve this, you should first increase `database_advanced.max_connections`, then stop `packetfence-mariadb` on all 3 servers, and follow the steps in the section 'Recovering from a split brain' of this document. Note that you can use any of the database servers as your source of truth.
+In the event that one of the 3 servers reaches the maximum amount of
+connections (defaults to 1000), this will dead-lock the Galera cluster
```
Comment (extrafu) [inline docs/cluster/appendix.asciidoc]: deadlock

File: docs/cluster/maintenance_and_operations.asciidoc [hunk 15]
```diff
@@ -18,13 +18,21 @@ endif::[]
 
 //== Maintenance and Operations
 
-=== Putting a node in maintenance
+=== Putting nodes in maintenance
 
-When doing maintenance on a node (especially the management node), it is
-always better to put it in maintenance mode so it doesn't try to join an
-existing cluster.
+When doing maintenance on a cluster, it is always better to put the targeted
```
Comment (extrafu) [inline docs/cluster/maintenance_and_operations.asciidoc]: better -> preferred 
to put -> to set

File: docs/cluster/maintenance_and_operations.asciidoc [hunk 16]
```diff
@@ -18,13 +18,21 @@ endif::[]
...
+nodes in a maintenance mode so they doesn't try to join an existing
```
Comment (extrafu) [inline docs/cluster/maintenance_and_operations.asciidoc]: they doesn't -> they don't

**Comment by extrafu** [comment]
All comments added.

**Comment by nqb** [comment]
Done.

**Pull request merged**